ondewo/t2s/text-to-speech.proto

```diff
@@ -19,34 +19,107 @@ import "google/protobuf/empty.proto";
 
 // endpoints of t2s generate service
 service Text2Speech {
+
+    // Synthesizes an specific text sent in the request with the configuration requirements and retrieves a response
+    // that includes the synthesized text to audio and the configuration wanted.
     rpc Synthesize (SynthesizeRequest) returns (SynthesizeResponse) {
     };
+    // will this safe time when doing batch predict on the AI model?
+    rpc BatchSynthesize (BatchSynthesizeRequest) returns (BatchSynthesizeResponse) {
+    };
+    // Retrieves the configuration of the specified pipeline.
     rpc GetT2sPipeline (T2sPipelineId) returns (Text2SpeechConfig) {
     };
+    // Creates a pipeline with the specified configuration and retrieves its id.
     rpc CreateT2sPipeline (Text2SpeechConfig) returns (T2sPipelineId) {
     };
+    // Deletes the specified pipeline.
     rpc DeleteT2sPipeline (T2sPipelineId) returns (google.protobuf.Empty) {
     };
+    // Update a specified pipeline with certain configuration.
     rpc UpdateT2sPipeline (Text2SpeechConfig) returns (google.protobuf.Empty) {
     };
+    // Retrieve the list of pipelines with an specific requirement.
     rpc ListT2sPipelines (ListT2sPipelinesRequest) returns (ListT2sPipelinesResponse) {
     };
+    // Retrieve the list of languages given a specific config request.
     rpc ListT2sLanguages (ListT2sLanguagesRequest) returns (ListT2sLanguagesResponse) {
     };
+    // Retrieve the list of domains given a specific config request.
     rpc ListT2sDomains (ListT2sDomainsRequest) returns (ListT2sDomainsResponse) {
     };
 }
 
 ////////////////
 // SYNTHESIZE //
 ////////////////
+
+// Represents a Synthesize Request.
+// A Synthesize Request contains the information need to perform a text to speech conversion.
 message SynthesizeRequest {
-    string t2s_pipeline_id = 1; // id of the model that will generate audio
-    string text = 2; // text for t2s
-    float length_scale = 3; // this parameter defines the speed of the audio. Should not be very much off 1.0
-    float noise_scale = 4; // can be from 0.0 to 1. Defines the noise in the generated audio
-    int32 sample_rate = 5; // by default 22050 sample rate of the generated wav file
-    enum Pcm {
+
+    // Required. Represents the text that will be transformed to speech.
+    string text = 1;
+
+    // Required. Represents the specifications needed to do the text to speech transformation.
+    RequestConfig config = 2;
+}
+
+message BatchSynthesizeRequest {
```

Comment:
I think it's not a good idea to have repeated fields for the text and the configs. What if something goes wrong, and you send more texts than configs (or vice-versa)? Why not just having one single field: 
_repeated SynthesizeRequest requests = 1;_